nodezator/fontsman/test_cache.py

```diff
@@ -0,0 +1,79 @@
+from unittest import TestCase
```

Comment:
For this file, may I please ask you to make a number of small changes before I can accept the pull request?

For each requested change listed below I provide an explanation as to why I'm requesting them.

Let me know if you disagree with anything, so we can discuss it further, since it is not my intention to mindlessly force people to do what I want without proper reasoning.

If you do agree but prefer that I do the changes myself, it is fine as well, just let me know.

And here are the changes:

#### Regarding the entire file

The usage of `assertIs()` should be replaced by `assertEqual()` whenever `assertIs()` is being used with literals. This is so because we should not use identity (`is`) to compare literals. Even the Python interpreter complains about it nowadays (from Python 3.10.6 on my machine):

```python
>>> 1 is 1
<stdin>:1: SyntaxWarning: "is" with a literal. Did you mean "=="?
True
```

This is so because literals might not be the same object always. It only happens that Python uses the same object for a literal sometimes (maybe oftentimes?), but it is not guaranteed, which is why the interpreter complains.

#### For `test_get_font()` and `test_fonts_db_returns()`

`self.assertIs(font.get_height(), height)`: this assertion should be removed altogether, because even if we replaced `assertIs` with `assertEqual`, the equality between `font.get_height()` and the requested height is not guaranteed. `pygame.font.Font.get_height()` returns an average height of all the glyphs in the font, which is not what we want, otherwise we'd use it directly inside our custom `get_font()` function instead of measuring the height of the rendered space character repeatedly inside the while-loop within the function.

`self.assertEqual(font.name, name)`: this is not wrong per se, but it is out of the scope of the operations within `nodezator/fontsman/cache.py`. That is, since we do not use this attribute of the pygame.font.Font instances, its existence and actual format/content is irrelevant to our usage. If the contents of this attribute changes in the future (perhaps in a new pygame version?), this assertion would make our test fail even though the actual code would be working 100% fine, that is, it would be a false positive indicating a potential problem in our code when in fact the problem would be in an assumption of the contents of an attribute that we do not even use.

#### For `test_get_font_too_large_raises()`

Both instances of `assertIs(len(db), 1)` should be replaced by `assertEqual(len(db), 1)`, since we are comparing literals, as previously explained.

Reply:
Thank you for finding so many issues that can be improved. I updated my PR.

* migrated `assertIs` -> `assertEqual`
* eliminated self.assertEqual(font.get_height(), height) 

also:

* renamed `raise_if_unattainable`, better matching existing names

I feel I am learning good Python programming practices with this PR - thank you!

Reply:
I'm learning as well.

I didn't know that multiple assignments in a single line had that disadvantage you pointed out regarding the lines 48 and 78 of the original `fontsman/cache.py` that you split into two lines each.

I was under the assumption that putting everything in the same line would avoid typos by reducing the number of places a variable has to be typed in files, thus helping in case I wanted to rename a variable for instance. And I guess this advantage does exist, but as demonstrated by your pull request, the risks are disproportionately high, as it can hide missing assignments, like the one that caused the performance issue.

I also think splitting such assignments into multiple lines makes it easier for control version software to keep track of changes as well, since it can track changes more accurately across multiple lines than when we are editing everything in the same line.